### DISCUSSION TITLE
Label creation for filters

### DISCUSSION THREAD

**Discussion by Antiokh** [martinrotter/rssguard#1034] [Ideas]
It's a great idea to have filters and labels, but sometimes I need to create a label.

In example, I have an RSS-post like:

```
The coolest event ever!

City: London
Price: 300$
```

And I don't want to create all-the cities before they mentioned in a RSS, I want to create them as labels when they are in the entry, assign the existing label or create new (and then assign).

**Comment by martinrotter** [comment]
It is possible, will respond in detail tomorrow.

**Comment by martinrotter** [comment]
Can you provide sample link to feed you want to process this way so that I can create sample filter?

**Reply by Antiokh** [reply]
[Atom](https://www.upwork.com/ab/feed/topics/atom?securityToken=f94c94e37bb8e7801e831b75357f1b7cfa48092188d86f5a8a707902ad2b1b19f44af48180f671825dfadd16fbde1e880a64b71e7ac89b1dd649c0e3eea50156&userUid=1555538926219706368&orgUid=1555538926219706369&topic=6660391)
[RSS](https://www.upwork.com/ab/feed/topics/rss?securityToken=f94c94e37bb8e7801e831b75357f1b7cfa48092188d86f5a8a707902ad2b1b19f44af48180f671825dfadd16fbde1e880a64b71e7ac89b1dd649c0e3eea50156&userUid=1555538926219706368&orgUid=1555538926219706369&topic=6660391)

I wanna catch the city and skills sections

**Reply by martinrotter** [reply]
OK. Here it is. I found some related bug in RSS Guard, but the feature works.

```js
function filterMessage() {
  // Deal with country.
  var country_rx = msg.contents.match(/<b>Country<\/b>: ?([\w ]+)/);

  if (country_rx != null && country_rx.length > 0) {
    var country = country_rx[1];
    var country_lbl = msg.createLabelId(`Country: ${country}`);

    msg.assignLabel(country_lbl);
  }

  // Deal with skills.
  var skills_rx = msg.contents.match(/<b>Skills<\/b>: ?([-\w, .]+)/);

  if (skills_rx != null && skills_rx.length > 0) {
    var skills_all = skills_rx[1];
    var skills = skills_all.split(/,? {2,5}/);

    skills.forEach((skl) => {
      if (skl != null && skl.length > 0) {
        var skill_lbl = msg.createLabelId(`Skill: ${skl}`);
        msg.assignLabel(skill_lbl);
      }
    });
  }

  return MessageObject.Accept;
}
```

![image](https://github.com/martinrotter/rssguard/assets/1255302/49044255-8219-40af-9fd1-80e2bd776a59)

**Comment by Antiokh** [comment]
Thank you!

I've tried this filter (added it and set corresponding feeds), but it doesn't work (I tried on Atom feeds it it's important).When I click "Test" on a filter it gives me the output: 

> EXISTING articles filtering error: 'TypeError: Property 'createLabelId' of object MessageObject(0x7ffe33a5acf0) is not a function'.
> Article will be ACCEPTED.
> 
> Output (modified) article is:
>   Title = 'Year of Linux Desktop'
>   URL = 'https://mynews.com/news/5'
>   Author = 'Napoleon Bonaparte'
>   Is read/important = 'no/yes'
>   Created on = '1692484482785'
>   Contents = '<p>Browsers usually insert quotation marks around the q element.</p><p>WWF's goal is to: <q>Build a future where people live in harmony with nature.</q></p>'
>   RAW contents = '<entry><title>Year of Linux Desktop</title><link href="https://mynews.com/news/5" rel="alternate" type="text/html" title="Year of Linux Desktop"/><published>2023-08-19T22:34:42</published><author><name>Napoleon Bonaparte</name></author><updated>2023-08-19T22:34:42</updated><id>https://mynews.com/news/5</id><summary type="html">&lt;p&gt;Browsers usually insert quotation marks around the q element.&lt;/p&gt;&lt;p&gt;WWF's goal is to: &lt;q&gt;Build a future where people live in harmony with nature.&lt;/q&gt;&lt;/p&gt;</summary></entry>'

BTW, it would be nice to have an ability to select the item from the feed and use its content as a base for test command.

UPD. I've noticed that my distribution of Ubuntu has RSS Guard 4.0.4. I'll update it and try again.

**Comment by Antiokh** [comment]
Thank you, now everything works fine. First time it created the duplicates for every tag, so I started from one RSS channel to create initial tags and added a duplicate filter from the standard ones. There was an issue where the ring bells every fetch, regardless if there are any new items, but it's gone in 4.5.
